apps/spinbyt/spinbyt.star

```diff
@@ -0,0 +1,225 @@
+"""
+Applet: Spinbyt
+Summary: Shows Spin scooters info
+Description: App that shows the nearest Spin scooter, its battery level, and number of other nearby scooters. Includes a scooter icon.
+Author: zachlucas
+"""
+
+load("render.star", "render")
+load("http.star", "http")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("math.star", "math")
+load("schema.star", "schema")
+
+DEFAULT_URL = "https://gbfs.spin.pm/api/gbfs/v2_3/%s/free_bike_status"
+
+DEFAULT_LOCATION = """
+{
+	"lat": "40.46049905272754",
+	"lng": "-79.95109706964388",
+	"description": "600 Grant St, Pittsburgh, PA, USA",
+	"locality": "Pittsburgh",
+	"place_id": "ChIJCSF8lBZEwokRhngABHRcdoI",
+	"timezone": "America/New_York"
+}
+"""
+
+DEFAULT_CITY = "pittsburgh"
+
+def main(config):
+    # City from configuration:
+    city = config.get("city", DEFAULT_CITY)
+    url = DEFAULT_URL % city
+
+    # Location (Lat/Lon) from configuration:
+    location = config.get("location", DEFAULT_LOCATION)
+    loc = json.decode(location)
+
+    # Simple scooter image:
+    scooter_image = base64.decode("iVBORw0KGgoAAAANSUhEUgAAABwAAAAcCAYAAAByDd+UAAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAHKADAAQAAAABAAAAHAAAAABkvfSiAAABO0lEQVRIDe2V0Q3CMAxEA2IMviokGIYdugd8lIpJwlR8dQNmKFzFBcdOWkErQIhIVZzkzi9OIXXuU8173+J5K/8d0BkrSlVXlmVYp25sn0yo4VOCk8C2Wnfv8lTsooKmAM+jjGogATJWsqeGBtjemjucQxKC9DEHwdgAwK7dj3VsPu03Fc5uzR02WjfZ2AAny5xJ9AdmDub16d8/0sXrh2OducuBlwccTwNzSS3+MQMPoVkgLm7v3agPMiCAbffHQDdACE5heTjgzqmsqiraJNYvy1WYiz5PgMHYNA39riiKLtaJg0AEGoaluq67Kuk3QAljLpgY53rCpDY1F/0PCYOJDwA05mB6vk8fAaWxzyR1qVhWqdfNj4aCPhM1socem9Qb1XnMu9EGJNUmCdKx9Kd8BogEQyYN+erxFYZIkUf9Flp2AAAAAElFTkSuQmCC")
+
+    # Make a GET request
+    rep = http.get(url)
```

Comment:
We'll need a bit of caching around this call to make sure we're not hammering the API too much. Every Tidbyt running the app will trigger one execution of `main()` each time the app is on display, so caching is essential to keep request rates down.